bin/commandTopicConsumer.py

```diff
@@ -0,0 +1,165 @@
+                        name = match.group(1).upper()
+                        if name == "KSQL_PROCESSING_LOG":
+                            continue
+                        if name not in stmts:
+                            stmts[name] = []
+                        stmts[name].append(record.stmt)
+                    else:
+                        match2 = re.search(r'(?:terminate) (?:ctas|csas)_(.+?)_', record.stmt, re.I)
+                        if match2:
+                            name = match2.group(1).upper()
+                            stmts[name].append(record.stmt)
+                        else:
+                            print("No match %s",record)
+                    # High watermark is +2 from last offset
+                    if msg.offset() >= max_offset-2:
+                        break
+
+                except ValueError as ex:
+                    print("consumer: Failed to deserialize message in "
+                                     "{} [{}] at offset {} (headers {}): {}".format(
+                                         msg.topic(), msg.partition(), msg.offset(), msg.headers(), ex))
+                    self.msg_err_cnt += 1
+
+        except KeyboardInterrupt:
+            pass
+
+        finally:
+            self.consumer.close()
+            print("Consumed {} messages, erroneous message = {}.".format(self.msg_cnt, self.msg_err_cnt))
+            for key, value in stmts.items():
+                print("[{}]".format(key))
+                for cmd in value:
+                    print("> {}".format(cmd))
+                print("-------------------------------------------------\n")
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Command Topic consumer client')
+    parser.add_argument('-b', dest='brokers', type=str, default=None, help='Bootstrap servers')
+    parser.add_argument('-f', dest='confFile', type=argparse.FileType('r'),
+                        help='Configuration file (configProp=value format)')
+    parser.add_argument('-k', dest='ksqlServiceId', type=str, default=None, help='KsqlDB service ID')
+
+    args = parser.parse_args()
+
+    conf = dict()
+    if args.confFile is not None:
+        # Parse client configuration file
+        for line in args.confFile:
+            line = line.strip()
+            if len(line) == 0 or line[0] == '#':
+                continue
+
+            i = line.find('=')
+            if i <= 0:
+                raise ValueError("Configuration lines must be `name=value..`, not {}".format(line))
+
+            name = line[:i]
+            value = line[i+1:]
+
+            conf[name] = value
+
+    if args.brokers is not None:
+        # Overwrite any brokers specified in configuration file with
+        # brokers from -b command line argument
+        conf['bootstrap.servers'] = args.brokers
+    elif 'bootstrap.servers' not in conf:
+        conf['bootstrap.servers'] = 'localhost:9092'
+
+
+    if 'group.id' not in conf:
```

Comment:
is there any reason not to always generate a group.id? we don't want to accidentally join a group that's doing real processing

Reply:
Why do we need a group id at all? If we just pass None to the consumer we can just consume without a group.

Reply:
It doesn't work if I specify None : 
```
cimpl.KafkaException: KafkaError{code=_UNKNOWN_GROUP,val=-179,str="Failed to set subscription: Local: Unknown group"}
```

Reply:
I think that's because you're setting up your partition assignment using `subscribe`, which joins a group. You can also just do an explicit assignment instead of calling `subscribe` by using `assign` (btw that's what our command topic consumer does.
): https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html#kafka.KafkaConsumer.assign

Ultimately the behavior is the same, so up to you.